posthog/hogql/database/database.py

```diff
@@ -247,9 +250,26 @@ def get_warehouse_tables(self) -> list[str]:
     def get_views(self) -> list[str]:
         return self._view_table_names
 
+    # This guaranttes that we can have both tables and views (or anything)
+    # with the same namespace (like Stripe, for example) and they're merged
+    # together as an attribute when we try setting them
+    def merge_or_setattr(self, f_name: str, f_def: Any):
+        current = getattr(self, f_name, None)
+        if current is not None:
+            if isinstance(current, TableGroup) and isinstance(f_def, TableGroup):
+                current.merge_with(f_def)  # Inplace
+            else:
+                raise ValueError(
+                    f"Conflict trying to add table {f_name}: {current} and {f_def} have the same key but are not the same"
+                )
+        else:
+            setattr(self, f_name, f_def)
+
+        return self
```

Comment:
@EDsCODE This is the bulk of the change. Rather than calling `setattr` on `stripe` twice, we're now calling this helper that merges the two `stripe` `TableGroup`s into one